internal/controller/util/misc.go

```diff
@@ -254,3 +260,28 @@ func TrimToK8sResourceNameLength(name string) string {
 
 	return name
 }
+
+func GetJobName(namePrefix string, ownerName string) string {
+	return getShortenedResourceName(namePrefix, ownerName, JobNameMaxLength)
+}
+
+func GetServiceName(namePrefix string, ownerName string) string {
+	return getShortenedResourceName(namePrefix, ownerName, ServiceNameMaxLength)
+}
+
+func getShortenedResourceName(namePrefix string, ownerName string, maxLength int) string {
```

Comment:
Maybe add volsync to the name of this function, since the logic is tied to volsync logic and we cannot change it. So this can be `volsyncShortenedResourceName()`.

Actually all these function should not be in utils/misc.go, they should be in the volsync package since they are needed only for volsync. If we need shorter resource names in other places we can use a better algorithm.